hw3/app/app/src/Search.jsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+
+function Search(props) {
+  const { title } = props;
+  const { characters } = props;
+  const [input, setInput] = useState('');
+  const [search, setSearch] = useState('');
+
+  const handleChange = (event) => {
+    setInput(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setSearch(input.toLowerCase());
+  };
+
+  function executeSearch() {
+    let results = characters.filter((character) => {
+      if (character.fullName.toLowerCase().includes(search)) {
+        return character;
+      }
+      return '';
```

Comment:
You might consider adding some sort of message to the user if their search returns no results.